pkg/controller/utils/datadog/metrics_forwarder_test.go

```diff
@@ -1029,3 +1029,49 @@ func Test_metricsForwarder_getSecretsFromCache(t *testing.T) {
 		})
 	}
 }
+
+func Test_getbaseURL(t *testing.T) {
```

Comment:
@ahmed-mez Added the unit test. 

However I thought of something while doing this. Typically, `ddUrl` would be a custom URL for the Datadog endpoint (`app.datadoghq.com`). However, the API client uses by default `api.datadoghq.com`. 

Therefore it might be best to only use Site and totally disregard `ddUrl` to set the API endpoint for the Metrics Forwarder. What do you think?

Reply:
Thanks, what you said is true most of the time the difference is either `.com` or `.eu` but in some advanced cases we might want to use a completely different url. So if `dda.Spec.Agent.Config.DDUrl` is set we need to use it to send operator metrics to the same backend as the agent. (the current logic in your PR)